indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -894,7 +885,12 @@ public TaskLocation getTaskLocation(final String id)
       @Override
       public Optional<TaskStatus> getTaskStatus(String id)
       {
-        return taskStorage.getStatus(id);
+        final Optional<TaskQueue> taskQueue = taskMaster.getTaskQueue();
```

Comment:
If the task is not present in the `taskQueue`, we should then check the `taskStorageQueryAdapter`. It might be already complete.